### PR TITLE
All the changes from BSU

### PR DESCRIPTION
**Pull request by prensing**

### CODE REVIEW

File: src/main/java/frc/robot/commands/GetCenterNoteX.java [hunk 1]
```diff
@@ -57,7 +57,7 @@ public GetCenterNoteX(Translation2d targetNote, DriveTrain driveTrain, NoteVisio
 
             // wait up to 1 second to suck the Note in all the way
             // new WaitUntilCommand(intake::hasNote).withTimeout(1),
-            new WaitCommand(1),
+            new WaitCommand(0.5),
```
Comment (jacobh1) [inline src/main/java/frc/robot/commands/GetCenterNoteX.java:60]: We might want to make this a named constant. We could do some testing how much to reduce it.

File: src/main/java/frc/robot/commands/GetStageNoteX.java [hunk 2]
```diff
@@ -38,7 +38,7 @@ public GetStageNoteX(Translation2d targetNote, DriveTrain driveTrain, NoteVision
 
             // wait up to 1 more second to suck the Note in all the way
             //new WaitUntilCommand(intake::hasNote).withTimeout(2),
-            new WaitCommand(1),
+            new WaitCommand(0.5),
```
Comment (jacobh1) [inline src/main/java/frc/robot/commands/GetStageNoteX.java:41]: make timeout value a constant

File: src/main/java/frc/robot/commands/StartIntake.java [hunk 3]
```diff
@@ -17,7 +17,7 @@ public class StartIntake extends SequentialCommandGroup {
     /** Creates a new startIntake. */
     public StartIntake(Intake intake, Shooter shooter, ShooterPivot pivot, Elevator elevator) {
         addCommands(
-                new Stow(shooter, pivot, elevator),
+                new Stow(shooter, pivot, elevator).withTimeout(0.75),
```
Comment (jacobh1) [inline src/main/java/frc/robot/commands/StartIntake.java:20]: make timeout value a constant

File: src/main/java/frc/robot/subsystems/Climber.java [hunk 4]
```diff
@@ -20,30 +20,36 @@ public class Climber extends SubsystemBase {
...
-    // private boolean m_rightHookComplete = false;
+    private boolean m_leftHookComplete = false;
+    private boolean m_rightHookComplete = false;
     private double m_rollAngle;
 
     // Constants to be used in this class
-    private static final double WINCH_GEAR_RATIO = 1.0/15.0;
-    private static final double NOMINAL_WINCH_DIAMETER = Units.inchesToMeters(0.75);
-    private static final double NOMINAL_INCHES_PER_ROTATION = Math.PI * NOMINAL_WINCH_DIAMETER * WINCH_GEAR_RATIO;
+    // private static final double WINCH_GEAR_RATIO = 1.0/15.0;
```
Comment (jacobh1) [inline src/main/java/frc/robot/subsystems/Climber.java:34]: Helpful to save the commented lines, or can we delete?

File: src/main/java/frc/robot/subsystems/Climber.java [hunk 5]
```diff
@@ -151,20 +170,25 @@ else if (m_climberState == ClimberState.RETRACTING_HOOKS) {
             // Once one side goes high, we need to stop that winch until the other hook "catches up"
             // Note that when we enetered this state, both winches started retracting. We only need to stop them one at a time until
             // the robot is level again and then we will go to the CLIMBING state
-            if (m_rollAngle > ROLL_ANGLE_TOLERANCE) {
+
+            // TODO: what about if they both engage at the same time and stay ~level?
```
Comment (jacobh1) [inline src/main/java/frc/robot/subsystems/Climber.java:174]: Can we capture this TODO in a more visible place, and decide if this needs work?

File: src/main/java/frc/robot/subsystems/DriveTrain.java [hunk 6]
```diff
@@ -53,9 +53,8 @@ public class DriveTrain extends SubsystemBase {
     // used in lots of places, so create a local constant
     private static final double MAX_VELOCITY_METERS_PER_SECOND = FalconDriveController.MAX_VELOCITY_METERS_PER_SECOND;
 
-    // TODO get correct values
-    public static final double PATH_PLANNER_MAX_VELOCITY = 2.0; //5.0;
-    public static final double PATH_PLANNER_MAX_ACCELERATION = 2.0; //5.0;
+    public static final double PATH_PLANNER_MAX_VELOCITY = 4.5; //5.0;
+    public static final double PATH_PLANNER_MAX_ACCELERATION = 3.5; //5.0;
```
Comment (jacobh1) [inline src/main/java/frc/robot/subsystems/DriveTrain.java:57]: We need to test:
 A) What the max speed/acceleration that we're comfortable with, that give a meaningful time benefit 
and 
B) Should all paths use the same values from these constants, or is it helpful in some cases to use path-specific velocity/acceleration values?

**Review by jacobh1 — APPROVED** [review]
Looks good to me. A few comments

File: src/main/java/frc/robot/subsystems/Climber.java (hunk 4 above)
Comment (prensing) [reply src/main/java/frc/robot/subsystems/Climber.java:34]: Think we should save for now.

File: src/main/java/frc/robot/subsystems/Climber.java (hunk 5 above)
Comment (prensing) [reply src/main/java/frc/robot/subsystems/Climber.java:174]: fixed comment

File: src/main/java/frc/robot/subsystems/DriveTrain.java (hunk 6 above)
Comment (prensing) [reply src/main/java/frc/robot/subsystems/DriveTrain.java:57]: Agreed

File: src/main/java/frc/robot/commands/ActiveTurnToHeadingWithDriving.java [hunk 7]
```diff
@@ -67,7 +67,7 @@ public void execute() {
             speed = -MathUtil.clamp(m_turnHeadingPID.calculate(headingDegrees, wantedDegrees), -1.0, 1.0);
         }
 
-        m_driveTrain.joystickDrive(m_joystickXSupplier.getAsDouble(), m_joystickYSupplier.getAsDouble(), speed);
+        m_driveTrain.joystickDrive(m_joystickXSupplier.getAsDouble(), m_joystickYSupplier.getAsDouble(), speed, false);
```
Comment (coachFitz0807) [inline src/main/java/frc/robot/commands/ActiveTurnToHeadingWithDriving.java:70]: Please add comment to explain the "false" parameter

File: src/main/java/frc/robot/commands/TurnToHeadingWithDriving.java [hunk 8]
```diff
@@ -56,7 +56,7 @@ public void initialize() {
   public void execute() {
     // auto aiming using PID
       double speed = m_turnHeadingPID.calculate(m_driveTrain.getHeading().getDegrees(), m_wantedDegrees);
```
Comment (coachFitz0807) [inline src/main/java/frc/robot/commands/TurnToHeadingWithDriving.java:58]: In ActiveSetShooterWithDriving, there's a clamp() around this to mmimit it to [-1.0,1.0]. Do we want that here? Or remove it from there?

**Review by coachFitz0807 — APPROVED** [review]

**Pull request merged**